packages/lambdas/src/document-upload.ts

```diff
@@ -0,0 +1,135 @@
+// import * as Sentry from "@sentry/serverless";
+// import status from "http-status";
+import { capture } from "./shared/capture";
+import { getEnvOrFail } from "./shared/env";
+import { S3Event } from "aws-lambda";
+import { makeS3Client } from "@metriport/core/external/aws/s3";
+import axios from "axios";
+import { getFileInfoFromS3 } from "./shared/file-info";
+import { MetriportError } from "@metriport/core/util/error/metriport-error";
+
+// // Keep this as early on the file as possible
+capture.init();
+
+const apiServerURL = getEnvOrFail("API_URL");
+// const destinationBucket = getEnvOrFail("MEDICAL_DOCUMENTS_BUCKET");
+const api = axios.create();
+const region = getEnvOrFail("AWS_REGION");
+const s3 = makeS3Client(region);
+
+const buildResponse = (status: number, body?: unknown) => ({
+  statusCode: status,
+  body,
+});
+
+type FileData = {
+  mimetype?: string;
+  size?: number;
+  docId: string;
+  cxId: string;
+  patientId: string;
+};
+
+export const handler = async (event: S3Event) => {
+  if (event.Records[0]) {
+    const sourceBucket = event.Records[0].s3.bucket.name;
+    const sourceKey = decodeURIComponent(event.Records[0].s3.object.key.replace(/\+/g, " "));
+    const destinationBucket = "devs.metriport.com";
+    const { destinationKey, queryParams } = getDestinationKeyAndQueryParams(sourceKey);
+
+    const params = {
+      CopySource: encodeURI(`${sourceBucket}/${sourceKey}`),
+      Bucket: destinationBucket,
+      Key: destinationKey,
+    };
+    // Make a copy of the file to the general medical documents bucket
+    try {
+      await s3.copyObject(params).promise();
+    } catch (err) {
+      console.log("Error copying file to general medical documents bucket", JSON.stringify(err));
+      capture.error("Error copying file to general medical documents bucket", {
+        extra: { context: `document-upload`, err, destinationKey, sourceKey },
+      });
+      throw err;
+    }
+
+    // Get file info from the copied file
+    const { size, contentType } = await getFileInfoFromS3(destinationKey, destinationBucket);
+    const fileData = {
+      size,
+      mimetype: contentType,
+      ...getIdsFromKey(destinationKey),
+    };
```

Comment:
Don't we need to send the S3 location/key to the API as well?

Reply:
Just noticed its being sent as queryParams... it would be better to send as part of the request body, no? Query params have length limitations and, since we're already sending data on the body, its easier/cleaner on the route to processs those, including validation w/ Zod schema